compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -525,20 +535,59 @@ convertModule lfVersion pkgMap stablePackages isGenerated file x depOrphanModule
           ]
         interfaceCons = interfaceNames lfVersion (eltsUFM (cm_types x))
         tplImplements = MS.fromListWith (++)
-           [ (mkTypeCon [getOccText tplTy], [ifaceTy])
-           | (name, _) <- binds
-           , "_implements_" `T.isPrefixOf` getOccText name
-           , TypeCon _ [TypeCon tplTy [], TypeCon ifaceTy []] <- [varType name]
-           ]
-        tplInterfaceInstances :: MS.Map (GHC.Module, TypeConName, TypeConName) (GHC.Expr GHC.CoreBndr)
-        tplInterfaceInstances = MS.fromList
-           [ ((mod, mkTypeCon [iface], mkTypeCon [getOccText tpl]), val)
-           | (name, val) <- binds
-           , DFunId _ <- [idDetails name]
-           , TypeCon ifaceCls [TypeCon tpl []] <- [varType name]
-           , Just iface <- [T.stripPrefix "Is" $ getOccText ifaceCls]
-           , Just mod <- [nameModule_maybe (getName ifaceCls)]
-           ]
+          [ (mkTypeCon [getOccText tpl], [iface])
+          | (name, _val) <- binds
+          , DFunId _ <- [idDetails name]
+          , TypeCon implementsCls [TypeCon tpl [], TypeCon iface []] <- [varType name]
+          , NameIn DA_Internal_Desugar "Implements" <- [implementsCls]
+          ]
+        tplInterfaceMethodInstances :: MS.Map (GHC.Module, TypeConName, TypeConName) [(T.Text, GHC.Expr GHC.CoreBndr)]
+        tplInterfaceMethodInstances = MS.fromListWith (++)
+          [
+            ( (mod, mkTypeCon [getOccText iface], mkTypeCon [getOccText tpl])
+            , [(methodName, body)]
+            )
+          | (name, val) <- binds
+          , TypeCon methodNewtype
+            [ TypeCon tpl []
+            , TypeCon iface []
+            , StrLitTy methodName
+            , _methodType
+            ] <- [varType name]
+          , NameIn DA_Internal_Desugar "Method" <- [methodNewtype]
+          , Just mod <- [nameModule_maybe (getName iface)]
+          , Var (NameIn DA_Internal_Desugar "mkMethod")
+            `App` Type _tpl
+            `App` Type _iface
+            `App` Type _methodName
+            `App` Type _methodType
+            `App` _implDict
+            `App` _hasMethodDict
+            `App` body
+              <- [untick val]
```

Comment:
If you noticed me frowning during the meeting, it was because of this! 

The problem was that in the `InterfaceDesugared` test, the rhss of the `_metod_<...>` bindings got an extra `Tick` node, so this list comprehension was returning the empty list

Reply:
Typical ghc mischief!